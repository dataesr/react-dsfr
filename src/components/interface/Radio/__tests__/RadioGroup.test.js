import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { v4 as uuidv4 } from 'uuid';
import { RadioGroup, Radio } from '../index';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('uuid', () => ({
  v4: jest.fn(),
}));

describe('<RadioGroup />', () => {
  beforeEach(() => {
    uuidv4.mockImplementationOnce(() => 'xxxxxxx');
  });
  it('renders correctly', () => {
    const component = renderer
      .create(
        <RadioGroup
          legend="Légende pour l’ensemble de champs"
          name="radio-test"
        >
          <Radio
            label="Label de la radio 1"
            value="value 1"
            data-testid="radio 1"
          />
          <Radio
            label="Label de la radio 2"
            value="value 2"
            data-test-id="radio 2"
          />
        </RadioGroup>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('works in controlled mode', () => {
    let value = '';
    const onChange = (newValue) => { value = newValue; };
    const component = mount(
      <RadioGroup
        legend="Radio Group"
        onChange={onChange}
      >
        <Radio
          label="Label 1"
          value="1"
          data-test-id="radioButton1"
        />
        <Radio
          label="Label 2"
          value="2"
          data-test-id="radioButton2"
        />
      </RadioGroup>,
    );

    const radio2 = component.find('[data-test-id="radioButton2"] input').first();
    radio2.simulate('change', { target: { checked: true } });

    expect(value).toBe('2');
  });
});
