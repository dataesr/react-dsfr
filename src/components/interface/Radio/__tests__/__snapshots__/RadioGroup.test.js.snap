// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioGroup /> renders correctly 1`] = `
<div
  className="fr-form-group"
>
  <fieldset
    aria-label="Légende pour l’ensemble de champs"
    className="fr-fieldset"
    disabled={false}
  >
    <legend
      className="fr-fieldset__legend fr-text--regular"
    >
      Légende pour l’ensemble de champs
    </legend>
    <div
      className="fr-fieldset__content"
    >
      <div
        className="fr-radio-group"
        data-testid="radio 1"
      >
        <input
          id="xxxxxxx"
          name="radio-test"
          onChange={[Function]}
          required={false}
          type="radio"
          value="value 1"
        />
        <label
          className="fr-label"
          htmlFor="xxxxxxx"
        >
          Label de la radio 1
        </label>
      </div>
      <div
        className="fr-radio-group"
        data-test-id="radio 2"
      >
        <input
          name="radio-test"
          onChange={[Function]}
          required={false}
          type="radio"
          value="value 2"
        />
        <label
          className="fr-label"
        >
          Label de la radio 2
        </label>
      </div>
    </div>
  </fieldset>
</div>
`;
